modules/extension/wmts/src/main/java/org/geotools/ows/wmts/model/WMTSService.java

```diff
@@ -14,20 +14,49 @@
  */
 package org.geotools.ows.wmts.model;
 
+import java.net.MalformedURLException;
+import java.net.URL;
 import java.util.ArrayList;
 import java.util.List;
 import net.opengis.ows11.LanguageStringType;
 import net.opengis.ows11.ServiceIdentificationType;
+import net.opengis.ows11.ServiceProviderType;
 import net.opengis.ows11.impl.KeywordsTypeImpl;
 import net.opengis.ows11.impl.LanguageStringTypeImpl;
 import org.geotools.data.ows.Service;
+import org.geotools.metadata.iso.citation.ResponsiblePartyImpl;
+import org.geotools.util.SimpleInternationalString;
 
 /**
  * @author ian
  * @author Matthias Schulze (LDBV at ldbv dot bayern dot de)
  */
 public class WMTSService extends Service {
 
+    public WMTSService(ServiceIdentificationType serviceType, ServiceProviderType serviceProvider) {
+
+        this(serviceType);
+
+        ResponsiblePartyImpl contactInfo =
```

Comment:
This needs a check against an empty serviceProvider. In the WMTS spec, the serviceProvder section is optional. Real life example is view-source:https://geodata.nationaalgeoregister.nl/tiles/service/wmts?request=GetCapabilities&service=WMTS